include/ioreq.h

```diff
@@ -35,8 +37,7 @@ typedef struct io_proxy {
     void (*backend_notify)(struct io_proxy *io_proxy);
     int (*run)(struct io_proxy *io_proxy);
     uint32_t backend_id;
-    uintptr_t data_base;
-    size_t data_size;
+    fdt_reserved_memory_t *data_plane;
```

Comment:
This effectively blocks using non-generated DTBs and configuration without SWIOTLB, and I think the biggest problem is that it couples configuration and operation.

I don't see a reason why this should be part of the io_proxy struct. We could use another struct which is f.ex. registered (or simply an array that is generated) and matched using `backend_id` when a node needs to be generated.

Reply:
How about now?